In [4]:
import pandas as pd
import numpy as np
import time

import seaborn as sn
import matplotlib.pyplot as plt



from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score
from sklearn.preprocessing import MinMaxScaler
from sklearn.model_selection import train_test_split
from sklearn.svm import SVC
from sklearn.naive_bayes import GaussianNB
from sklearn.neighbors import KNeighborsClassifier

import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense
from tensorflow import keras
import tensorflow as tf


import warnings
from sklearn.exceptions import ConvergenceWarning
warnings.filterwarnings("ignore", category=ConvergenceWarning)
# Re-enable the warning
#warnings.filterwarnings("default", category=ConvergenceWarning)

In [5]:
df = pd.read_csv('train.csv',index_col = 'id')

## Checking the Data

In [6]:
count_nan = df.isnull().sum()
count_nan

BertzCT              0
Chi1                 0
Chi1n                0
Chi1v                0
Chi2n                0
Chi2v                0
Chi3v                0
Chi4n                0
EState_VSA1          0
EState_VSA2          0
ExactMolWt           0
FpDensityMorgan1     0
FpDensityMorgan2     0
FpDensityMorgan3     0
HallKierAlpha        0
HeavyAtomMolWt       0
Kappa3               0
MaxAbsEStateIndex    0
MinEStateIndex       0
NumHeteroatoms       0
PEOE_VSA10           0
PEOE_VSA14           0
PEOE_VSA6            0
PEOE_VSA7            0
PEOE_VSA8            0
SMR_VSA10            0
SMR_VSA5             0
SlogP_VSA3           0
VSA_EState9          0
fr_COO               0
fr_COO2              0
EC1                  0
EC2                  0
EC3                  0
EC4                  0
EC5                  0
EC6                  0
dtype: int64

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 14838 entries, 0 to 14837
Data columns (total 37 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   BertzCT            14838 non-null  float64
 1   Chi1               14838 non-null  float64
 2   Chi1n              14838 non-null  float64
 3   Chi1v              14838 non-null  float64
 4   Chi2n              14838 non-null  float64
 5   Chi2v              14838 non-null  float64
 6   Chi3v              14838 non-null  float64
 7   Chi4n              14838 non-null  float64
 8   EState_VSA1        14838 non-null  float64
 9   EState_VSA2        14838 non-null  float64
 10  ExactMolWt         14838 non-null  float64
 11  FpDensityMorgan1   14838 non-null  float64
 12  FpDensityMorgan2   14838 non-null  float64
 13  FpDensityMorgan3   14838 non-null  float64
 14  HallKierAlpha      14838 non-null  float64
 15  HeavyAtomMolWt     14838 non-null  float64
 16  Kappa3             14838 no

In [8]:
df.describe()

BertzCT          Chi1         Chi1n         Chi1v         Chi2n  \
count  14838.000000  14838.000000  14838.000000  14838.000000  14838.000000   
mean     515.153604      9.135189      5.854307      6.738497      4.432570   
std      542.456370      6.819989      4.647064      5.866444      3.760516   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%      149.103601      4.680739      2.844556      2.932842      1.949719   
50%      290.987941      6.485270      4.052701      4.392859      2.970427   
75%      652.652585     11.170477      7.486791      8.527859      5.788793   
max     4069.959780     69.551167     50.174588     53.431954     32.195368   

              Chi2v         Chi3v         Chi4n   EState_VSA1   EState_VSA2  \
count  14838.000000  14838.000000  14838.000000  14838.000000  14838.000000   
mean       5.253221      3.418749      1.773472     29.202823     10.435316   
std        4.925065      3.436208      1.865898     31.728679     13.651843   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        2.034468      1.160763      0.503897      5.969305      0.000000   
50%        3.242775      1.948613      1.073261     17.353601      6.420822   
75%        6.609350      4.502070      2.534281     44.876559     12.841643   
max       34.579313     22.880836     16.072810    363.705954     99.936429   

       ...    SlogP_VSA3   VSA_EState9        fr_COO       fr_COO2  \
count  ...  14838.000000  14838.000000  14838.000000  14838.000000   
mean   ...     13.636941     49.309959      0.458215      0.459226   
std    ...     14.598554     29.174824      0.667948      0.668111   
min    ...      0.000000     -5.430556      0.000000      0.000000   
25%    ...      4.794537     30.000000      0.000000      0.000000   
50%    ...      9.589074     41.666667      0.000000      0.000000   
75%    ...     14.912664     56.090650      1.000000      1.000000   
max    ...    115.406157    384.450519      8.000000      8.000000   

                EC1           EC2           EC3           EC4           EC5  \
count  14838.000000  14838.000000  14838.000000  14838.000000  14838.000000   
mean       0.667745      0.798962      0.313789      0.279081      0.144831   
std        0.471038      0.400790      0.464047      0.448562      0.351942   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.000000      1.000000      0.000000      0.000000      0.000000   
50%        1.000000      1.000000      0.000000      0.000000      0.000000   
75%        1.000000      1.000000      1.000000      1.000000      0.000000   
max        1.000000      1.000000      1.000000      1.000000      1.000000   

                EC6  
count  14838.000000  
mean       0.151570  
std        0.358616  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        0.000000  
max        1.000000  

[8 rows x 37 columns]

In [9]:
df.drop(['EC3', 'EC4', 'EC5', 'EC6'],axis = 1,inplace = True)

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 14838 entries, 0 to 14837
Data columns (total 33 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   BertzCT            14838 non-null  float64
 1   Chi1               14838 non-null  float64
 2   Chi1n              14838 non-null  float64
 3   Chi1v              14838 non-null  float64
 4   Chi2n              14838 non-null  float64
 5   Chi2v              14838 non-null  float64
 6   Chi3v              14838 non-null  float64
 7   Chi4n              14838 non-null  float64
 8   EState_VSA1        14838 non-null  float64
 9   EState_VSA2        14838 non-null  float64
 10  ExactMolWt         14838 non-null  float64
 11  FpDensityMorgan1   14838 non-null  float64
 12  FpDensityMorgan2   14838 non-null  float64
 13  FpDensityMorgan3   14838 non-null  float64
 14  HallKierAlpha      14838 non-null  float64
 15  HeavyAtomMolWt     14838 non-null  float64
 16  Kappa3             14838 no

In [11]:
df.describe()

BertzCT          Chi1         Chi1n         Chi1v         Chi2n  \
count  14838.000000  14838.000000  14838.000000  14838.000000  14838.000000   
mean     515.153604      9.135189      5.854307      6.738497      4.432570   
std      542.456370      6.819989      4.647064      5.866444      3.760516   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%      149.103601      4.680739      2.844556      2.932842      1.949719   
50%      290.987941      6.485270      4.052701      4.392859      2.970427   
75%      652.652585     11.170477      7.486791      8.527859      5.788793   
max     4069.959780     69.551167     50.174588     53.431954     32.195368   

              Chi2v         Chi3v         Chi4n   EState_VSA1   EState_VSA2  \
count  14838.000000  14838.000000  14838.000000  14838.000000  14838.000000   
mean       5.253221      3.418749      1.773472     29.202823     10.435316   
std        4.925065      3.436208      1.865898     31.728679     13.651843   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        2.034468      1.160763      0.503897      5.969305      0.000000   
50%        3.242775      1.948613      1.073261     17.353601      6.420822   
75%        6.609350      4.502070      2.534281     44.876559     12.841643   
max       34.579313     22.880836     16.072810    363.705954     99.936429   

       ...     PEOE_VSA7     PEOE_VSA8     SMR_VSA10      SMR_VSA5  \
count  ...  14838.000000  14838.000000  14838.000000  14838.000000   
mean   ...     11.318811      6.704487     15.666766     31.066423   
std    ...     20.169745     10.865415     18.080208     33.896638   
min    ...      0.000000      0.000000      0.000000      0.000000   
25%    ...      0.000000      0.000000      5.969305      6.420822   
50%    ...      0.000000      0.000000     11.752550     20.075376   
75%    ...     13.847474      6.923737     17.721856     42.727765   
max    ...    211.501279    100.348416     80.742293    492.729739   

         SlogP_VSA3   VSA_EState9        fr_COO       fr_COO2           EC1  \
count  14838.000000  14838.000000  14838.000000  14838.000000  14838.000000   
mean      13.636941     49.309959      0.458215      0.459226      0.667745   
std       14.598554     29.174824      0.667948      0.668111      0.471038   
min        0.000000     -5.430556      0.000000      0.000000      0.000000   
25%        4.794537     30.000000      0.000000      0.000000      0.000000   
50%        9.589074     41.666667      0.000000      0.000000      1.000000   
75%       14.912664     56.090650      1.000000      1.000000      1.000000   
max      115.406157    384.450519      8.000000      8.000000      1.000000   

                EC2  
count  14838.000000  
mean       0.798962  
std        0.400790  
min        0.000000  
25%        1.000000  
50%        1.000000  
75%        1.000000  
max        1.000000  

[8 rows x 33 columns]

# Training Models

In [12]:
df.columns

Index(['BertzCT', 'Chi1', 'Chi1n', 'Chi1v', 'Chi2n', 'Chi2v', 'Chi3v', 'Chi4n',
       'EState_VSA1', 'EState_VSA2', 'ExactMolWt', 'FpDensityMorgan1',
       'FpDensityMorgan2', 'FpDensityMorgan3', 'HallKierAlpha',
       'HeavyAtomMolWt', 'Kappa3', 'MaxAbsEStateIndex', 'MinEStateIndex',
       'NumHeteroatoms', 'PEOE_VSA10', 'PEOE_VSA14', 'PEOE_VSA6', 'PEOE_VSA7',
       'PEOE_VSA8', 'SMR_VSA10', 'SMR_VSA5', 'SlogP_VSA3', 'VSA_EState9',
       'fr_COO', 'fr_COO2', 'EC1', 'EC2'],
      dtype='object')

In [13]:
X = df.iloc[:, 0:31]
y = df.iloc[:,31:33]
X_train, X_test,y_train, y_test = train_test_split(X,y , random_state=104,test_size=0.25,shuffle=True)
y1 = y_train['EC1']
y2 = y_train['EC2']
x1 = y_test['EC1']
x2 = y_test['EC2']
Time = []
Time.append("")

In [14]:
X.info()

<class 'pandas.core.frame.DataFrame'>
Index: 14838 entries, 0 to 14837
Data columns (total 31 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   BertzCT            14838 non-null  float64
 1   Chi1               14838 non-null  float64
 2   Chi1n              14838 non-null  float64
 3   Chi1v              14838 non-null  float64
 4   Chi2n              14838 non-null  float64
 5   Chi2v              14838 non-null  float64
 6   Chi3v              14838 non-null  float64
 7   Chi4n              14838 non-null  float64
 8   EState_VSA1        14838 non-null  float64
 9   EState_VSA2        14838 non-null  float64
 10  ExactMolWt         14838 non-null  float64
 11  FpDensityMorgan1   14838 non-null  float64
 12  FpDensityMorgan2   14838 non-null  float64
 13  FpDensityMorgan3   14838 non-null  float64
 14  HallKierAlpha      14838 non-null  float64
 15  HeavyAtomMolWt     14838 non-null  float64
 16  Kappa3             14838 no

## Linear Regression

In [15]:
start = time.time()
#EC1
modelL1 = LogisticRegression(C=0.5, penalty='l2', solver='liblinear', max_iter=100)
modelL1.fit(X_train, y1)
predictions = modelL1.predict(X_test)
accuracyL1 = accuracy_score(x1, predictions)
#EC2
modelL2 = LogisticRegression( penalty='l2', solver='liblinear', max_iter=1000)
modelL2.fit(X_train, y2)
predictions = modelL2.predict(X_test)
accuracyL2 = accuracy_score(x2, predictions)
print("Accuracy1:", accuracyL1)
print("Accuracy2:", accuracyL2)
end = time.time()
finalL = (end-start) * 10**3
print("The time of execution of above program is :",finalL, "ms")
Time.append(finalL)

Accuracy1: 0.6870619946091644
Accuracy2: 0.8037735849056604
The time of execution of above program is : 1565.9706592559814 ms


## SVC

In [16]:
start = time.time()

#EC1
modelS1 = SVC()
modelS1.fit(X_train, y1)
predictions1 = modelS1.predict(X_test)
accuracyS1 = accuracy_score(x1, predictions1)
#EC2
modelS2 = SVC()
modelS2.fit(X_train, y2)
predictions2 = modelS2.predict(X_test)
accuracyS2 = accuracy_score(x2, predictions2)
print("Accuracy1:", accuracyS1)
print("Accuracy2:", accuracyS2)
end = time.time()
finalS = (end-start) * 10**3
print("The time of execution of above program is :",finalS, "ms")
Time.append(finalS)

Accuracy1: 0.6803234501347709
Accuracy2: 0.8032345013477089
The time of execution of above program is : 13324.904441833496 ms


In [17]:
X_train

BertzCT       Chi1     Chi1n      Chi1v     Chi2n     Chi2v  \
id                                                                      
8140   602.980459  11.170522  5.976884   5.976884  5.381308  5.381308   
13029  272.131680   4.599171  2.813865   2.813865  2.093756  2.093756   
11407  529.312414   7.194306  4.267026   5.797322  3.381003  4.620356   
9213   263.261606   6.502908  4.441467   3.972030  2.797412  2.797412   
4474   952.679420  10.810968  7.142063  12.662235  5.392377  8.229500   
...           ...        ...       ...        ...       ...       ...   
8846   195.127041   6.308862  3.414884   3.414884  2.757262  2.757262   
13563  203.028307   6.275874  3.599704   5.377157  2.861666  3.990568   
729    133.774438   4.680739  2.316497   2.316497  1.316497  1.316497   
5825   731.665451  10.806897  7.599423  10.966475  5.336400  7.640696   
8261    29.019550   2.642734  1.236744   1.236744  0.639972  0.639972   

          Chi3v     Chi4n  EState_VSA1  EState_VSA2  ...  PEOE_VSA14  \
id                                                   ...               
8140   3.054553  2.072890    29.929472    17.584700  ...   11.938611   
13029  1.120915  0.213865    29.901842     0.000000  ...   11.938611   
11407  2.640965  1.305666    44.938685     6.420822  ...   13.792002   
9213   1.708684  0.965398    27.005766    19.623859  ...   85.618586   
4474   5.153471  1.770579    52.351343     6.420822  ...   15.645394   
...         ...       ...          ...          ...  ...         ...   
8846   1.819833  1.027595     0.000000    11.499024  ...    0.000000   
13563  1.929821  1.037112     0.000000     0.000000  ...   85.618586   
729    0.610305  0.000000    17.353601    12.841643  ...    5.969305   
5825   6.612498  3.203726    50.632613    16.981741  ...   13.792002   
8261   0.000000  0.000000     5.969305     5.573105  ...    5.969305   

       PEOE_VSA6  PEOE_VSA7  PEOE_VSA8  SMR_VSA10   SMR_VSA5  SlogP_VSA3  \
id                                                                         
8140    0.000000   0.000000   0.000000  11.938611  24.925325   14.383612   
13029   0.000000   0.000000  12.173675  11.938611  18.883484    9.589074   
11407   0.000000   0.000000   6.420822  19.479389  24.856656   13.825658   
9213    0.000000   6.066367   0.000000  19.479389   6.103966   18.349406   
4474    0.000000   0.000000   6.420822  32.757611  24.601926   22.701338   
...          ...        ...        ...        ...        ...         ...   
8846   18.218407  36.010352   0.000000  10.969244  36.623798    0.000000   
13563   0.000000   0.000000   0.000000  17.744066  43.870935    4.794537   
729     6.923737  13.344559   0.000000   5.969305   6.420822    9.589074   
5825    0.000000   0.000000   6.923737  19.479389  37.567504   13.825658   
8261    0.000000   0.000000   0.000000   5.969305   6.041841    4.794537   

       VSA_EState9  fr_COO  fr_COO2  
id                                   
8140     90.335681       2        2  
13029    53.333333       2        2  
11407    55.905811       2        2  
9213     31.554606       0        0  
4474     53.196840       0        0  
...            ...     ...      ...  
8846     34.416667       0        0  
13563    29.653794       0        0  
729      34.166667       1        1  
5825     42.467877       0        0  
8261     23.333333       1        1  

[11128 rows x 31 columns]

## Random Forest:

In [ ]:
start = time.time()
#EC1

rf_classifier = RandomForestClassifier()
# Set the hyperparameters
rf_classifier.n_estimators = 10000  # Number of decision trees
rf_classifier.criterion = 'gini'  # Splitting criterion
rf_classifier.min_samples_split = 2
rf_classifier.fit(X_train, y1)
predictions1 = rf_classifier.predict(X_test)
accuracyR1 = accuracy_score(x1, predictions1)

rf_classifier2 = RandomForestClassifier()
  # Set the hyperparameters
rf_classifier2.n_estimators = 10000  # Number of decision trees
rf_classifier2.criterion = 'gini'  # Splitting criterion
rf_classifier2.min_samples_split = 2
rf_classifier2.fit(X_train, y2)
predictions2 = rf_classifier2.predict(X_test)
accuracyR2 = accuracy_score(x2, predictions2)
print("Accuracy1:", accuracyR1)
print("Accuracy2:", accuracyR2)
end = time.time()
finalR = (end-start) * 10**3
print("The time of execution of above program is :",finalR, "ms")
Time.append(finalR)

## Xg BOOSt

In [ ]:
start = time.time()

#EC1
modelG1 = xgb.XGBClassifier()
modelG1.fit(X_train, y1)
predictions1 = modelG1.predict(X_test)
accuracyG1 = accuracy_score(x1, predictions1)
#EC2
modelG2 = xgb.XGBClassifier()
modelG2.fit(X_train, y2)
predictions2 = modelG2.predict(X_test)
accuracyG2 = accuracy_score(x2, predictions2)
print("Accuracy1:", accuracyG1)
print("Accuracy2:", accuracyG2)

end = time.time()
finalG = (end-start) * 10**3
print("The time of execution of above program is :",finalG, "ms")
Time.append(finalG)

## Naive Bayes:

In [ ]:
start = time.time()

#EC1
modelN1 = GaussianNB()
modelN1.fit(X_train, y1)
predictions1 = modelN1.predict(X_test)
accuracyN1 = accuracy_score(x1, predictions1)
#EC2
modelN2 = GaussianNB()
modelN2.fit(X_train, y1)
predictions2 = modelN2.predict(X_test)
accuracyN2 = accuracy_score(x2, predictions2)
print("Accuracy1:", accuracyN1)
print("Accuracy2:", accuracyN2)

end = time.time()
finalN = (end-start) * 10**3
print("The time of execution of above program is :",finalN, "ms")
Time.append(finalN)

## K-Nearest Neighbors (KNN):

In [ ]:
start = time.time()
#EC1
modelK1 = KNeighborsClassifier()
modelK1.fit(X_train, y1)
predictionK1 = modelK1.predict(X_test)
accuracyK1 = accuracy_score(x1, predictionK1)
#EC2
modelK2 = KNeighborsClassifier()
modelK2.fit(X_train, y2)
predictions2 = modelK2.predict(X_test)
accuracyK2 = accuracy_score(x2, predictions2)
print("Accuracy1:", accuracyK1)
print("Accuracy2:", accuracyK2)
end = time.time()
finalK = (end-start) * 10**3
print("The time of execution of above program is :",finalK, "ms")
Time.append(finalK)

## Tensorflow logistic regression

In [ ]:
start = time.time()
modelT1 = keras.Sequential([
    keras.layers.Dense(64, activation='relu', input_shape=(31,)),
    keras.layers.Dense(64, activation='relu'),
    keras.layers.Dense(1, activation='sigmoid')
])


# Compile the model
modelT1.compile(optimizer='adam',
              loss='binary_crossentropy',
              metrics=['accuracy'])

# Train the model
modelT1.fit(X_train, y1, epochs=100, verbose=1)

# Evaluate the model on the test set
loss, accuracyT1 = modelT1.evaluate(X_test, x1, verbose=0)
print("Test Loss:", loss)
print("Test Accuracy:", accuracyT1)

y_pred = modelT1.predict(X_test)
y_pred_classes = np.round(y_pred).flatten()

In [ ]:

modelT2 = keras.Sequential([
    keras.layers.Dense(64, activation='relu', input_shape=(31,)),
    keras.layers.Dense(64, activation='relu'),
    keras.layers.Dense(1, activation='sigmoid')
])

# Compile the model
modelT2.compile(optimizer='adam',
              loss='binary_crossentropy',
              metrics=['accuracy'])

# Train the model
modelT2.fit(X_train, y2, epochs=100, verbose=1)

# Evaluate the model on the test set
loss, accuracyT2 = modelT2.evaluate(X_test, x2, verbose=0)
print("Test Loss:", loss)
print("Test Accuracy:", accuracyT2)

y_pred = modelT2.predict(X_test)
y_pred_classes = np.round(y_pred).flatten()

end = time.time()
finalT = (end-start) * 10**3
print("The time of execution of above program is :",finalT, "ms")
Time.append(finalT)

## Gradient Boosting

In [ ]:
# Import models and utility functions
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
from sklearn.datasets import load_digits

# Setting SEED for reproducibility
SEED = 23
# Instantiate Gradient Boosting Regressor
gbc1 = GradientBoostingClassifier(n_estimators=10000,
                                    learning_rate=0.05,
                                    random_state=100,
                                    max_features=5 )
gbc1.fit(X_train, y1)
# Predict on test set
pred_y = gbc1.predict(X_test)
acc1 = accuracy_score(x1, pred_y)
print("Gradient Boosting Classifier accuracy is : {:.2f}".format(acc1))


In [ ]:
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
from sklearn.datasets import load_digits
# Setting SEED for reproducibility
SEED = 23
# Instantiate Gradient Boosting Regressor
gbc2 = GradientBoostingClassifier(n_estimators=10000,
    learning_rate=0.05,
    random_state=100,
    max_features=5 )
gbc2.fit(X_train, y2)
# Predict on test set
pred_y = gbc2.predict(X_test)
acc2 = accuracy_score(x2, pred_y)
print("Gradient Boosting Classifier accuracy is : {:.2f}".format(acc2))

## Bagging classifier

In [20]:
from sklearn import model_selection
from sklearn.ensemble import BaggingClassifier
from sklearn.tree import DecisionTreeClassifier
import pandas as pd

# load the data
seed = 8
kfold = model_selection.KFold(n_splits = 3,
                        random_state = None)


rf_classifier = RandomForestClassifier()
# Set the hyperparameters
rf_classifier.n_estimators = 1000  # Number of decision trees
rf_classifier.criterion = 'gini'  # Splitting criterion
rf_classifier.min_samples_split = 2


# no. of base classifier
num_trees = 10

# bagging classifier
modelB1 = BaggingClassifier(base_estimator = rf_classifier,
                            n_estimators = num_trees,
                            random_state = seed)
modelB1.fit(X_train, y1)

results = model_selection.cross_val_score(modelB1,X_train, y1, cv = kfold)
print("accuracy :")
print(results.mean())


C:\Users\Varun Chindage\AppData\Roaming\Python\Python39\site-packages\sklearn\ensemble\_base.py:166: FutureWarning: `base_estimator` was renamed to `estimator` in version 1.2 and will be removed in 1.4.
  warnings.warn(
C:\Users\Varun Chindage\AppData\Roaming\Python\Python39\site-packages\sklearn\ensemble\_base.py:166: FutureWarning: `base_estimator` was renamed to `estimator` in version 1.2 and will be removed in 1.4.
  warnings.warn(
C:\Users\Varun Chindage\AppData\Roaming\Python\Python39\site-packages\sklearn\ensemble\_base.py:166: FutureWarning: `base_estimator` was renamed to `estimator` in version 1.2 and will be removed in 1.4.
  warnings.warn(
C:\Users\Varun Chindage\AppData\Roaming\Python\Python39\site-packages\sklearn\ensemble\_base.py:166: FutureWarning: `base_estimator` was renamed to `estimator` in version 1.2 and will be removed in 1.4.
  warnings.warn(


accuracy :
0.7024617277078145


In [21]:
from sklearn import model_selection
from sklearn.ensemble import BaggingClassifier
from sklearn.tree import DecisionTreeClassifier
import pandas as pd



seed = 8
kfold = model_selection.KFold(n_splits = 3,
                        random_state = None)

rf_classifier2 = RandomForestClassifier()
  # Set the hyperparameters
rf_classifier2.n_estimators = 1000  # Number of decision trees
rf_classifier2.criterion = 'gini'  # Splitting criterion
rf_classifier2.min_samples_split = 2



# no. of base classifier
num_trees = 10

# bagging classifier
modelB2 = BaggingClassifier(base_estimator = rf_classifier2,
                            n_estimators = num_trees,
                            random_state = seed)
modelB2.fit(X_train, y2)
results2 = model_selection.cross_val_score(modelB2,X_train, y2, cv = kfold)
print("accuracy :")
print(results2.mean())


C:\Users\Varun Chindage\AppData\Roaming\Python\Python39\site-packages\sklearn\ensemble\_base.py:166: FutureWarning: `base_estimator` was renamed to `estimator` in version 1.2 and will be removed in 1.4.
  warnings.warn(
C:\Users\Varun Chindage\AppData\Roaming\Python\Python39\site-packages\sklearn\ensemble\_base.py:166: FutureWarning: `base_estimator` was renamed to `estimator` in version 1.2 and will be removed in 1.4.
  warnings.warn(
C:\Users\Varun Chindage\AppData\Roaming\Python\Python39\site-packages\sklearn\ensemble\_base.py:166: FutureWarning: `base_estimator` was renamed to `estimator` in version 1.2 and will be removed in 1.4.
  warnings.warn(
C:\Users\Varun Chindage\AppData\Roaming\Python\Python39\site-packages\sklearn\ensemble\_base.py:166: FutureWarning: `base_estimator` was renamed to `estimator` in version 1.2 and will be removed in 1.4.
  warnings.warn(


accuracy :
0.7974490064113976


## Graph

In [ ]:
import plotly.graph_objects as go

# Sample data
categories = ['LR', 'SVC', 'RF', 'XG','NB','KNN','TF','GB','BC']
values = [accuracyL1, accuracyS1, accuracyR1, accuracyG1,accuracyN1,accuracyK1,accuracyT1,acc1,results]
# Create the bar trace
bar_trace = go.Bar(x=categories, y=values)

# Create the layout
layout = go.Layout(
    title='EC1',
    xaxis=dict(title='Model'),
    yaxis=dict(title='accuracy')
)

# Create the figure
fig = go.Figure(data=[bar_trace], layout=layout)

# Display the plot
fig.show()


In [ ]:
import plotly.graph_objects as go

# Sample data
# Sample data
categories = ['LR', 'SVC', 'RF', 'XG','NB','KNN','TF','GB','BC']
values = [accuracyL2, accuracyS2, accuracyR2, accuracyG2,accuracyN2,accuracyK2,accuracyT2,acc2,results2]

# Create the bar trace
bar_trace = go.Bar(x=categories, y=values)

# Create the layout
layout = go.Layout(
    title='EC2',
    xaxis=dict(title='Model'),
    yaxis=dict(title='accuracy')
)

# Create the figure
fig = go.Figure(data=[bar_trace], layout=layout)
fig.update_traces(textfont_size=12, textangle=0, textposition="outside", cliponaxis=False)
# Display the plot
fig.show()


In [ ]:

categories = ['LR', 'SVC', 'RF', 'XG','NB','KNN','TF']


bar_trace = go.Bar(x=categories, y=Time)


layout = go.Layout(
    title='Time',
    xaxis=dict(title='Model'),
    yaxis=dict(title='Time')
)

fig = go.Figure(data=[bar_trace], layout=layout)

fig.show()



As we can see  that allmost all the model have almost same accuarcy so we go for the second best parameter that is time, So the model with best time is:RF for EC1 and  LR for EC2

# Tables



1.   EC1:Accuracy1: 0.7080862533692722

2.   EC2:Accuracy2: 0.8037735849056604



In [22]:
pf = pd.read_csv('test.csv',index_col = 'id')
pf1 = pd.read_csv('test.csv')
pf1

id      BertzCT       Chi1      Chi1n      Chi1v      Chi2n  \
0     14838   344.632371   7.283603   4.473966   5.834958   3.412257   
1     14839  1432.410201  10.663869   7.079026   8.065215   5.297097   
2     14840    83.352608   3.931852   1.774215   1.774215   1.073446   
3     14841   150.255712   5.912790   3.548812   3.548812   2.595128   
4     14842  1817.276351  24.910940  15.540529  20.047314  12.535886   
...     ...          ...        ...        ...        ...        ...   
9888  24726   246.422865   4.036581   2.816709   2.816709   1.875634   
9889  24727   591.069706   8.770857   5.682461   5.682461   4.050440   
9890  24728   378.113435   6.310349   3.402334   4.317724   2.817428   
9891  24729   737.653518   9.949161   7.337949   7.337949   4.428511   
9892  24730   785.394062  12.170505   7.565385   9.651755   5.842572   

          Chi2v      Chi3v     Chi4n  EState_VSA1  ...  PEOE_VSA14  PEOE_VSA6  \
0      4.651530   2.096558  1.116433    49.458581  ...   13.512441   0.000000   
1      5.297097   3.924155  2.569694     0.000000  ...    0.000000  34.947374   
2      1.073446   0.467830  0.170838     5.969305  ...    5.969305   0.000000   
3      2.595128   1.642813  0.694113     0.000000  ...   59.935299   0.000000   
4     17.730988  11.979618  4.431173    84.554972  ...   23.468091  25.609359   
...         ...        ...       ...          ...  ...         ...        ...   
9888   1.875634   1.235986  0.362743    24.146543  ...   11.938611   0.000000   
9889   4.050440   2.167855  1.770579     0.000000  ...    0.000000  18.199101   
9890   4.071978   1.970236  1.165747    36.705949  ...    7.822697   0.000000   
9891   5.948361   3.972459  2.160881    36.992053  ...    0.000000   0.000000   
9892   8.229500   5.664696  2.587586    49.458581  ...    7.822697   0.000000   

      PEOE_VSA7  PEOE_VSA8  SMR_VSA10   SMR_VSA5  SlogP_VSA3  VSA_EState9  \
0      0.000000   0.000000  26.809272  24.539800    4.794537    47.304082   
1     98.323987   9.606882   0.000000  53.378235    0.000000    43.166667   
2      0.000000   6.420822  11.752550  13.344559    9.589074    24.666667   
3      0.000000   0.000000  17.744066  32.290168    4.794537    26.778866   
4      0.000000  37.099000  69.141353  38.704130   50.697492   102.583333   
...         ...        ...        ...        ...         ...          ...   
9888   0.000000   0.000000  11.938611  12.207933    9.589074    30.000000   
9889  37.107112  17.696186  10.969244   0.000000    0.000000    53.166667   
9890   0.000000   0.000000   7.822697  24.099010    4.736863    50.652870   
9891   0.000000   6.196844   0.000000  12.462662    9.589074    50.250000   
9892   0.000000   0.000000  18.872230  24.539800   13.825658    69.130957   

      fr_COO  fr_COO2  
0          1        1  
1          0        0  
2          1        1  
3          0        0  
4          0        0  
...      ...      ...  
9888       2        2  
9889       0        0  
9890       0        0  
9891       0        0  
9892       0        0  

[9893 rows x 32 columns]

In [ ]:
#EC1
EC1 = modelB1.predict(pf)
#EC2
EC2 = modelB2.predict(pf)
EC2

In [ ]:
sf = pd.read_csv('sample_submission.csv',index_col = 'id')

sf['EC1'] = EC1
sf['EC2'] = EC2
sf

In [ ]:
# saving the DataFrame as a CSV file
gfg_csv_data = sf.to_csv('SEND13.csv', index = True)
print('\nCSV String:\n', gfg_csv_data)